capstone-frontend/src/components/SearchUserModal.jsx

```diff
@@ -90,35 +169,52 @@ export class SearchUserModal extends Component {
 
   removeUserFromAddedUsers = (user) => {
 
-    const indexAddedUsers = this.state.addedUsers.indexOf(user);
-    this.state.addedUsers.splice(indexAddedUsers, 1)
-
-    if (!this.state.addedFriends.includes(user)) {
-      const indexAddedUsersTracker = this.state.addedUsersTracker.indexOf(user);
-      this.state.addedUsersTracker.splice(indexAddedUsersTracker, 1)
+    if (this.props.type === 'booklists' && this.props.bookList) {
+      this.removeUserFromBookListCollaborators(this.props.bookList, user);
     }
 
-    // Rerender
-    this.setState({ addedUsers: this.state.addedUsers, addedUsersTracker: this.state.addedUsersTracker })
-
+    if (!this.arrayContainsJSONId(this.state.addedFriends, user)) {
+      
+      // Rerender
+      this.setState(
+        {
+          addedUsers: this.state.addedUsers.filter(addedUser => addedUser.id !== user.id),
+          addedUsersTracker: this.state.addedUsersTracker.filter(addedUser => addedUser.id !== user.id),
+        }
+      );
+    }
+    else {
+      // Rerender
+      this.setState(
+        {
+          addedUsers: this.state.addedUsers.filter(addedUser => addedUser.id !== user.id),
+        }
+      );
```

Comment:
It looks like you are doing the same addedUser setState in both cases. Is it possible to have the first if check only set the `addedUsersTracker state` then remove the else and set the `addedUsers state` in all cases?

Reply:
This would cause react to do two re-render because ```setState``` could possibly be called twice. Is it okay if it re-renders twice? I feel like doing it once it better, even if it leads to duplicate code.

Reply:
I think this would probably be a better question for @geoffbass because I'm not sure exactly, I just know it appears to be cleaner code

Reply:
If you'd like to make sure you're just calling `setState` once, how about creating variables to hold `addedUsers` and `addedUsersTracker`? You can update the variables accordingly inside the conditional `if` and `else` branches (`addedUsersTracker` won't be updated in the `else` branch). You can finally call `setState` with both `addedUsers` and `addedUsersTracker` at the end outside the conditionals.